echopype/mask/api.py

```diff
@@ -33,9 +33,33 @@ def _validate_source_ds(source_ds, storage_options_ds):
         # open up Dataset using source_ds path
         source_ds = xr.open_dataset(source_ds, engine=file_type, chunks={}, **storage_options_ds)
 
-    # Check source_ds coordinates
-    if "ping_time" not in source_ds or "range_sample" not in source_ds:
-        raise ValueError("'source_ds' must have coordinates 'ping_time' and 'range_sample'!")
+    # Grab dimensions of mask
+    if isinstance(mask, list):
+        mask_dims = set()
+        for mask_indiv in mask:
+            for dim in mask_indiv.dims:
+                mask_dims.add(dim)
+    else:
+        mask_dims = set(mask.dims)
+
+    # Grab dimensions of the target variable in source ds
+    target_variable_dims = set(source_ds[var_name].dims)
+
+    # Raise ValueError if the mask has channel dim but the target variable doesn't
+    if "channel" in mask_dims and "channel" not in target_variable_dims:
+        raise ValueError(
+            "'channel' is in Mask dimensions but not in the Source " "Dataset Variable dimensions."
+        )
```

Comment:
What is the case when the target variable does not have a channel dimension? For all echopype-generated dataarrays I think we always have a channel dimension, since it start from the EchoData object?

Reply:
If we are keeping there's seems some quotation marks mismatches? Also thinking we can use a simpler string.
```suggestion
    # Raise ValueError if the mask has channel dim but the target variable doesn't
    if "channel" in mask_dims and "channel" not in target_variable_dims:
        raise ValueError(
            "'channel' is a dimension in mask but not a dimension in source."
        )
```

Reply:
> What is the case when the target variable does not have a channel dimension? For all echopype-generated dataarrays I think we always have a channel dimension, since it start from the EchoData object?

It was just in the case where somebody chose to do something like `ds_MVBS.isel(channel=0).drop("channel")` and then they chose to put in a mask created with the original number of channels that the `ds_MVBS` DataArray had before dropping channel. Maybe I am being too paranoid about that and this is too fringe (and non-sensical) a case 😆